Minesweeper.py

```diff
@@ -44,132 +35,133 @@ def __init__(self):
             win.blit(text, text_rect)
-        # DHB: I'm pretty certain only the top-level draw function should call display.update 
-        pygame.display.update() # Update the display
 
     def handle_events(self, events):
+        """ causes the transition from the tutorial state to the game state
+        """
         for event in events:
             if event.type == pygame.KEYDOWN:
                 if event.key == pygame.K_SPACE:
-                    # Transition to the game state
-                    # DHB: game is a separate object. one should never modify the contents of member variables
-                    # DHB: directly on another object. Instead call a function such as "set_state".
                     game.state = "game"
 
-class GameWonScreen(object): # creates the victory screen
+
+class GameWonScreen(object):
+    """ displays the game won screen
+    """
     def __init__(self):
-        self.font = pygame.font.Font(None, 30) # Font for win text
-        self.text_color = Color.BLACK # Color for win text
-        # DHB: instructions is not a reasonable variable name here
-        self.instructions = [
+        """ contains the text info to use in the draw function
+        """
+        self.font = pygame.font.Font(None, 30)
+        self.text_color = Color.BLACK
+        self.message = [
             "Congratulations! You win!",
             "Would you like to play again?",
             "",
-            "Press R to restart"
+            "Press R to restart",
+            "Press ESCAPE to quit"
         ]
 
-    def draw(self): # draws the win screen
-        # DHB: see comments elsewhere regarding fill and update
+    def draw(self):
+        """ draws the game won screen using the information in the init function
+        """
         win.fill(Color.BACKGROUND)
-        for i, instruction in enumerate(self.instructions):
+        for i, instruction in enumerate(self.message):
             text = self.font.render(instruction, True, self.text_color)
-            # DHB: generally its wise to not put too much in the same line. Here, I'd move the calculations to separate variables
-            # DHB: e.g.
-            #      x = win.get_width() // 2
-            #      y = 100 + i * 30
-            # DHB: while the other way works just fine, working in this fashion leads to bugs
             text_rect = text.get_rect(center=(win.get_width() // 2, 100 + i * 30))
             win.blit(text, text_rect)
 
-        pygame.display.update() # Update the display
+        pygame.display.update()
 
     def handle_events(self, events):
+        """ restarts the program when R is pressed and quits the program when ESCAPE is pressed
+        """
         for event in events:
             if event.type == pygame.KEYDOWN:
                 if event.key == pygame.K_r:
-                    # restart to the game state
                     game.restart()
+                if event.key == pygame.K_ESCAPE:
+                    run = False
```

Comment:
changing a local variable, "run" to False, has no effect on the overall game. Unless it's a global variable, which it better not be. Perhaps call instead something like game.exit(), and make that cause the program to halt (e.g. storing the state in a member variable and checking it in your main loop).